text/2019-08-21-supported-platforms.md

```diff
@@ -0,0 +1,230 @@
+efforts, while still inviting others to help us improve support.
+
+For production deployments, our recommendations (Tier 1) remain the same, and
+will remain limited to only a small selection of targets we have a long history
+with. While TiKV is compatible with many targets, our community members have
+years of experience maintaining critical workloads on these tier 1 targets and
+feel comfortable tackling the most in-depth problems. On other (tier 2) targets
+there may be differing best practices about how to manage services, how to
+control system limits, or other factors which may not be well documented. We are
+hopeful that over time we can build a sufficient knowledge base with other
+targets to adopt them as tier 1 targets.
+
+Still, we must recognize that while some platforms are either not well tested
+(Mac OS X), or completely untested (Windows), folks still *work* on those
+machines. Folks working with projects that talk to TiKV will want to have a copy
+of TiKV they can access and use. Folks working on TiKV will probably not be
+running a copy of their preferred deployment platform. These are *development*
+targets, where users benefit from being able to harness tooling or IDE features
+that require the TiKV source to at least *build* and be able to handle a couple
+operations a second. So it makes sense to define a second, less rigorous
+*Development platform* with tiers as well.
+
+A good example of this is the Nix package manager. While we are quite certain
+TiKV can operate on in Nix, our community has not yet buiilt TiKV with Nix. So
+we cannot be sure that Nix works, and it is not a platform we can recommend to
+use for development or deployment. If a community member added support, then
+TiKV could easily say that Nix is a perfectly reasonable development platform
+immediately. In the deployment platform case, even saying we gaurantee Nix to
+build (Tier 2) could take several months before our team is confident enough to
+provide official builds.
+
+## Detailed design
+
+This RFC proposes the following terms to be defined for TiKV:
+
+* **Deployment Platforms:** Deployment targets for release binaries produced by
+  the build process. These platforms can *run* TiKV, but not necessarily *build*
+  it. This platform is the concern of end user community.
+* **Development Platforms:** Development targets for interactively developing
+  TiKV on. These platforms can build, debug, and run TiKV, but may not have all
+  features to run a production workload. These platforms are only the concern of
+  our development community.
+* **Tier 1:** Official build processes build, test, and benchmark these targets,
+  and they are considered endorsed by the maintainers for all usage.
+      + Example: CentOS 7, our official build OS, on X86-64 with SSE4
+      + Full instructions present and regularly tested
+      + Tested on each PR
+      + Binaries provided
+* **Tier 2:** Official build processes build, but *do not require testing or
```

Comment:
"Official build processes build" I wouldn't say this, it means we'd be obliged to build for these platforms on CI, which we probably shouldn't do